protos/core.proto

```diff
@@ -260,24 +267,414 @@ message CompiledGraph {
+    // overrides applied.
+    // This option cannot be manually specified in action configs.
+    Target canonical_target = 2;
+
+    // Targets (non-canonical) of actions that this action is dependent on.
+    repeated Target dependency_targets = 3;
+
+    // Path to the source file that the contents of the action is loaded from.
+    string source = 4;
+
+    // Path that the config of the action is loaded from.
+    // This option cannot be manually specified in action configs.
+    string config_path = 5;
+
+    // A list of user-defined tags with which the action should be labeled.
+    repeated string tags = 6;
+
+    // If set to true, this action will not be executed. However, the action may
+    // still be depended upon.
+    bool disabled = 7;
+
+    // The SQL query.
+    string query = 10;
+
+    // Pre/post operations.
+    repeated string pre_ops = 11;
+    repeated string post_ops = 12;
+
+    // Common BigQuery options.
+    BigQueryOptions bigquery = 13;
+
+    // Applies the materialized view optimization, see
+    // https://cloud.google.com/bigquery/docs/materialized-views-intro.
+    bool materialized = 14;
+
+    // Description of the view.
+    string description = 15;
+
+    // Descriptions of columns within the table.
+    repeated ColumnDescriptor columns = 16;
+
+    // Key-value pairs for BigQuery labels. If the label name contains special
+    // characters, e.g. hyphens, then quote its name, e.g. `labels: {
+    // "label-name": "value" }`.
+    map<string, string> labels = 17;
+
+    // Key-value pairs for the table, view, and materialized view options.
+    map<string, string> additional_options = 18;
+  }
+
+  message IncrementalTable {
+    // Target is the unique action identifier.
+    Target target = 1;
+
+    // Canonical Target is the unique action identifier, with no target
+    // overrides applied.
+    // This option cannot be manually specified in action configs.
+    Target canonical_target = 2;
+
+    // Targets (non-canonical) of actions that this action is dependent on.
+    repeated Target dependency_targets = 3;
+
+    // Path to the source file that the contents of the action is loaded from.
+    string source = 4;
+
+    // Path that the config of the action is loaded from.
+    // This option cannot be manually specified in action configs.
+    string config_path = 5;
+
+    // A list of user-defined tags with which the action should be labeled.
+    repeated string tags = 6;
+
+    // If set to true, this action will not be executed. However, the action may
+    // still be depended upon.
+    bool disabled = 7;
+
+    // The SQL query.
+    string query = 8;
+
+    // The query that is applied when the table already exists.
+    // This option cannot be manually specified in action configs.
+    string incremental_query = 9;
+
+    // Pre/post operations.
+    repeated string pre_ops = 10;
+    repeated string post_ops = 11;
```

Comment:
nit: AFAIUI this is _also_ being used to define `actions.yaml`'s schema. if so, I think we should expand these to `pre_operations` etc